x/ibc/core/04-channel/types/msgs.go

```diff
@@ -164,6 +169,9 @@ func (msg MsgChannelOpenAck) ValidateBasic() error {
 	if err := host.ChannelIdentifierValidator(msg.ChannelId); err != nil {
 		return sdkerrors.Wrap(err, "invalid channel ID")
 	}
+	if err := host.ChannelIdentifierValidator(msg.CounterpartyChannelId); err != nil {
```

Comment:
shouldn't this be
```
	if msg.CounterpartyChannelId != "" {
		if err := host.ChannelIdentifierValidator(msg.CounterpartyChannelId); err != nil {
```

Reply:
no, `CounterpartyChannelId` must be set. The `!= ""` check in the handshake is against the stored counterparty channel id not the provided one

Reply:
By the time Ack rolls around, a connection ID should have been selected. The `""` stored string in the handshake just indicates that the selection was flexible and the counterparty channel id should be used as the counterparty's channel id